code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -15,239 +152,216 @@
+	create_full = TRUE
 
-/obj/item/flamethrower/proc/toggle_igniter(mob/user)
-	if(!ptank)
-		to_chat(user, "<span class='notice'>Attach a plasma tank first!</span>")
-		return
-	if(!status)
-		to_chat(user, "<span class='notice'>Secure the igniter first!</span>")
-		return
-	to_chat(user, "<span class='notice'>You [lit ? "extinguish" : "ignite"] [src]!</span>")
-	lit = !lit
+/obj/item/flamethrower/basic/full/tank
+	create_with_tank = TRUE
+
+
+/obj/item/flamethrower/advanced
+	name = "Burner Boyes Flamethrower"
+	desc = "Adapted from specialised salavaging equipment, the Burner Boyes flamerthrower lets you roast the average spaceman or cut through them."
+	icon = 'icons/obj/burner.dmi'
+	icon_state = "burner0"
+	item_state = "burner0"
+	lefthand_file = 'icons/mob/inhands/guns_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/guns_righthand.dmi'
+	flags = CONDUCT
+	force = 15 //it's heavy
+	armour_penetration = 0
+	throwforce = 10
+	throw_speed = 1
+	throw_range = 2 //again heavy
+	w_class = WEIGHT_CLASS_BULKY //not stuffing this into a backpack
+	materials = list(MAT_METAL=500)
+	resistance_flags = FIRE_PROOF
+	origin_tech = "combat=1;plasmatech=2;engineering=2"
+	flame_intensity = 1.4
+	consumption_rate = 20
+	tank_exposure = 3
+	lit = FALSE
+	tool_behaviour = null
+	toolspeed = 0.5
+	usesound = 'sound/items/welder.ogg'
+	actions_types = list(/datum/action/item_action/toggle_flamethrowermode)
+	var/consumption_rate_welder = 5
+	var/force_welder = 30
+	var/armour_penetration_welder = 50
+	var/status = FLAMETHROWERMODE
+	var/create_with_tank = FALSE
+
+/obj/item/flamethrower/advanced/Destroy()
+	QDEL_NULL(ptank)
+	return ..()
+
+/obj/item/flamethrower/advanced/update_icon()
+	cut_overlays()
+	if(ptank)
+		add_overlay("+ptank")
+		if(ptank.air_contents.toxins > 20) //preempting that runetime.
+			add_overlay("+fuel5")
+		else if(ptank.air_contents.toxins > 15)
+			add_overlay("+fuel4")
+		else if(ptank.air_contents.toxins > 10)
+			add_overlay("+fuel3")
+		else if(ptank.air_contents.toxins > 5)
+			add_overlay("+fuel2")
+		else if(ptank.air_contents.toxins < 5)
+			add_overlay("+fuel1")
+	else 
+		add_overlay("+fuel0")
 	if(lit)
-		START_PROCESSING(SSobj, src)
-		if(!warned_admins)
-			message_admins("[ADMIN_LOOKUPFLW(user)] has lit a flamethrower.")
-			warned_admins = TRUE
+		item_state = "burner[status]"
+		add_overlay("+lit[status]")
 	else
-		STOP_PROCESSING(SSobj,src)
-	update_icon()
+		item_state = "burner0"
+		icon_state = "burner[status]"
+	if(ismob(loc))
+		var/mob/M = loc
+		M.update_inv_l_hand()
+		M.update_inv_r_hand()
 
-/obj/item/flamethrower/CheckParts(list/parts_list)
+/obj/item/flamethrower/advanced/attack(mob/living/M, mob/living/user)
+	if(user.get_inactive_hand())
+		to_chat(user, "<span class='userdanger'>You need both hands free to attack with \the [src]!</span>")
+		return
+	if(lit && status == WELDERMODE) //we want it to have different attack values depending on what status and if it's on or not
+		damtype = BURN
```

Comment:
Move these to the toggle_mode and the toggle_igniter procs. No need to reevaluate them on each attack. Except for the consume_fuel ofc